lib/linter/linter.js

```diff
@@ -1713,17 +1708,67 @@ class Linter {
 
                         try {
 
-                            const ruleOptions = Array.isArray(ruleValue) ? ruleValue : [ruleValue];
+                            let ruleOptions = Array.isArray(ruleValue) ? ruleValue : [ruleValue];
 
-                            assertIsRuleOptions(ruleId, ruleValue);
```

Comment:
Note: I removed this because at this point `ruleValue` is always an array, so this wasn't actually checking anything. 

https://github.com/eslint/eslint/blob/8e13a6beb587e624cc95ae16eefe503ad024b11b/lib/config/flat-config-schema.js#L173-L177